processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -625,46 +626,85 @@ else if ( accessorNaming.isGetterMethod( candidate ) || Executables.isFieldAcces
      */
     private Accessor getAdderForType(Type collectionProperty, String pluralPropertyName) {
 
-        List<Accessor> candidates = new ArrayList<Accessor>();
-        if ( collectionProperty.isCollectionType ) {
-
-            // this is a collection, so this can be done always
-            TypeMirror typeArg = first( collectionProperty.determineTypeArguments( Iterable.class ) ).getTypeBound()
-                .getTypeMirror();
-            // now, look for a method that
-            // 1) starts with add,
-            // 2) and has typeArg as one and only arg
-            List<Accessor> adderList = getAdders();
-            for ( Accessor adder : adderList ) {
-                ExecutableElement executable = adder.getExecutable();
-                if ( executable == null ) {
-                    // it should not be null, but to be safe
-                    continue;
-                }
-                VariableElement arg = executable.getParameters().get( 0 );
-                if ( typeUtils.isSameType( arg.asType(), typeArg ) ) {
-                    candidates.add( adder );
-                }
-            }
+        List<Accessor> candidates;
+
+        if ( collectionProperty.isCollectionType() ) {
+            candidates = getAccessorCandidates( collectionProperty, Iterable.class );
+        }
+        else if ( collectionProperty.isStreamType() ) {
+            candidates = getAccessorCandidates( collectionProperty, getClassForName( JavaStreamConstants.STREAM_FQN ) );
```

Comment:
Here you can use `Stream.class` as the type is a `Stream` which means that the `Stream` is present. And It is safe to have that as on Java 6 and 7 the condition will never evaluate to `true` and thus it will never throw the `ClassNotFoundException`. You can then remove the `getClassFormName` method

Reply:
There is currently no import of ```java.util.stream.Stream``` (or any other Java 8 class) in the ```Type``` class. Using ```Stream.class``` here would introduce a Java 8 import. That was my reasoning behind this change. If you think the Java 8 import would not cause a problem, I am happy to revert this (including the ```getClassForName()``` method).